### PR TITLE
Cleanup MINI/NANO modifiers

### PR DESCRIPTION
**Pull request by vlimant**
#### PR description:

while looking for the proper modifier/era, in preparation for the 10.6 AOD reMINI-reNANO we stumbled upon many outdated process modifier and therefore cleaning them out

- run2_miniAOD_80XLegacy
- run2_miniAOD_94XFall17
- run2_miniAOD_devel
- run3_miniAOD_12X
- ~~miniAOD_UL_preSummer20~~

### CODE REVIEW

**Comment by vlimant** [comment]
is there an actual use of `run2_miniAOD_80XLegacy` `run2_miniAOD_94XFall17` ?

**Comment by vlimant** [comment]
interesting enough, there exists `Configuration/ProcessModifiers/python/run2_miniAOD_UL_cff.py` already with a similar purpose

# This modifier is for common settings to run miniAOD on top of 
# ultra-legacy (during LS2) Run-2 AOD

@cms-sw/orp-l2  what is the actual way to implement this : in Eras or ProcessModifiers ?

**Comment by vlimant** [comment]
could not attend last week ORP, although I left a question there.
@cms-sw/orp-l2  is there a way forward here : Eras or ProcessModifiers ?

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
type egm

**Comment by vlimant** [comment]
type jetmet

**Comment by vlimant** [comment]
type egamma

**Comment by vlimant** [comment]
please test

File: RecoEgamma/EgammaPhotonProducers/python/reducedEgamma_cfi.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import FWCore.ParameterSet.Config as cms
 
 from RecoEgamma.EgammaIsolationAlgos.interestingEgammaIsoDetIdsSequence_cff import interestingEgammaIsoHCALSel
+from RecoJets.Configuration.CaloTowersES_cfi import *
```
Comment (vlimant) [inline RecoEgamma/EgammaPhotonProducers/python/reducedEgamma_cfi.py:4]: this is very suspicious to me @cms-sw/egamma-pog-l2 can you please have someone look into this ?

Comment (afiqaize) [reply RecoEgamma/EgammaPhotonProducers/python/reducedEgamma_cfi.py:4]: test locally succeeds with only
```
import Geometry.CaloEventSetup.caloTowerConstituents_cfi
from Configuration.Eras.Modifier_phase2_hgcal_cff import phase2_hgcal
CaloTowerConstituentsMapBuilder = Geometry.CaloEventSetup.caloTowerConstituents_cfi.caloTowerConstituents.clone(
    SkipHE = True
)
```

**Comment by vlimant** [comment]
enable nano

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
please test
error must be spurious

**Comment by vlimant** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by hqucms** [comment]
@vlimant I think we still need `run2_miniAOD_UL_preSummer20` to read the data AODs for re-MINI, e.g., the following will crash if changing `run2_miniAOD_UL_preSummer20` to `run2_miniAOD_UL`:
```
cmsDriver.py step2  -s PAT --era Run2_2016_HIPM -n 100 --process PAT --conditions auto:run2_data --data  --scenario pp --eventcontent MINIAOD --datatier MINIAOD --procModifiers run2_miniAOD_UL_preSummer20 --filein /store/data/Run2016C/SingleMuon/AOD/21Feb2020_UL2016_HIPM-v1/20000/000B0CF4-2D3D-0442-9F7F-3D6FE3648F90.root --fileout file:step2.root
```

**Comment by vlimant** [comment]
what is in `run2_miniAOD_UL_preSummer20` that is not in `run2_miniAOD_UL` should be transferred over

**Comment by hqucms** [comment]
> what is in `run2_miniAOD_UL_preSummer20` that is not in `run2_miniAOD_UL` should be transferred over

```python
run2_miniAOD_UL_preSummer20 = cms.ModifierChain(run2_miniAOD_UL, miniAOD_skip_trackExtras)
```

And there are several modifications using `miniAOD_skip_trackExtras`.

**Comment by vlimant** [comment]
please test

**Comment by srimanob** [comment]
+Upgrade

**Comment by AdrianoDee** [comment]
+pdmv

**Comment by vlimant** [comment]
can @cms-sw/dqm-l2 ans @cms-sw/operations-l2 please sign or comment ?

**Comment by tjavaid** [comment]
+1

**Comment by rappoccio** [comment]
+1

**Pull request merged**